junit5-support/src/main/kotlin/in/specmatic/test/reports/OpenApiCoverageReportProcessor.kt

```diff
@@ -0,0 +1,64 @@
+package `in`.specmatic.test.reports
+
+import `in`.specmatic.core.ReportConfiguration
+import `in`.specmatic.core.ReportFormatterType
+import `in`.specmatic.core.log.logger
+import `in`.specmatic.test.reports.coverage.OpenAPICoverageReport
+import `in`.specmatic.test.reports.coverage.OpenApiCoverageReportInput
+import `in`.specmatic.test.reports.renderers.CoverageReportTextRenderer
+import `in`.specmatic.test.reports.renderers.ReportRenderer
+import org.assertj.core.api.Assertions.assertThat
+
+class OpenApiCoverageReportProcessor(private val openApiCoverageReportInput: OpenApiCoverageReportInput ): ReportProcessor {
+
+    override fun process(reportConfiguration: ReportConfiguration) {
+        openApiCoverageReportInput.addExcludedAPIs(reportConfiguration.types.apiCoverage.openAPI.excludedEndpoints)
+        val openAPICoverageReport = openApiCoverageReportInput.generate()
+        if (openAPICoverageReport.rows.isEmpty()) {
+            logger.log("The Open API coverage report generated is blank")
```

Comment:
Any reason this can happen? should we add what is the corrective action?

Reply:
Yes, this can happen in two scenarios:
1. All endpoints are added to the 'excludedEndpoints' array in the 'report' section in specmatic.json.
2. The open api spec does not have any paths documented and the 'endpointsApi' property is not set.

I have updated the message to:
```
The Open API coverage report generated is blank.
This can happen if you have included all the endpoints in the 'excludedEndpoints' array in the report section in specmatic.json, or if your open api specification does not have any paths documented.
```